src/main/java/com/google/api/codegen/transformer/php/PhpModelTypeNameConverter.java

```diff
@@ -174,11 +173,11 @@ private TypeName getTypeName(ProtoElement elem, int maxDepth) {
       }
 
       String parentFullName = getTypeName(parent, maxDepth - 1).getFullName();
-      String nickname = elem.getSimpleName();
-      String fullName = String.format("%s_%s", parentFullName, nickname);
+      String fullName = String.format("%s_%s", parentFullName, elem.getSimpleName());
+      String[] nameSplit = fullName.split("\\\\");
```

Comment:
Since you need only the last element something like the following may fit better (also allows to avoid double-escaping of the backslash):
```java
String nickName = fullName.substring(fullName.lastIndexOf("\\") + 1);
```

Reply:
Done